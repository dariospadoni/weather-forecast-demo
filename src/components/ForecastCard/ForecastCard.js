import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';
import './ForecastCard.scss';
import getIcon from '../WeatherIcons';
import { toCelsius } from '../../utils';


const ForecastCard = ({ selected, icon, date, temperature }) => 
  <article className={`forecast__card ${selected ? 'forecast--active': ''}`}>
    <div className="forecast__card__time">
      {moment(date).format('HH:mm')}
    </div>
    <div className="forecast__card__condition">
      {getIcon(icon)}
      </div>
    <div className="forecast__card__temperature">
      {toCelsius(temperature)}
    </div>
  </article>;


ForecastCard.propTypes = {
  selected: PropTypes.bool,
  date: PropTypes.instanceOf(Date).isRequired,
  temperature: PropTypes.number.isRequired,
  icon: PropTypes.string.isRequired,
};


export default ForecastCard;
