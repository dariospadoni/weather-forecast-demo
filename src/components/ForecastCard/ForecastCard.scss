.forecast__card {
  display: flex;
  flex-direction: column;
  align-items: center;
  color: white;
  padding: 0 24px;
  margin-right: 24px;
  border-radius: 6px;

  &.forecast--active {
    background-color: #51557A;
  }

  &:hover {
    background-color: #3B3F69;
    cursor: pointer;
  }

  & .forecast__card__time {
    color: #A8AABD;
    font-size: 48px;
    line-height: 56px;
    padding-top: 32px;
  }

  & .forecast__card__condition {
    padding-top: 32px;
    
    & svg {
      width: 120px;
      height: 120px;
    }
  }

  & .forecast__card__temperature {
    font-weight: 700;
    font-size: 82px;
    line-height: 96px;
    padding-top: 32px;
  }
}
