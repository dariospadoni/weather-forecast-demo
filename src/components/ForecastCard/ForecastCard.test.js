import React from 'react';
import { render } from '@testing-library/react';
import ForecastCard from './ForecastCard';
 
it('renders correctly when it is not selected', () => {
  const { container } = render(<ForecastCard date={new Date('2020-08-04 18:00:00')} icon={'01d'} temperature={294.05} selected={false} />);
  expect(container.firstChild.classList.contains('forecast--active')).toBe(false);
  expect(container).toMatchSnapshot();
});

it('renders correctly when selected', () => {
  const { container }  = render(<ForecastCard date={new Date('2020-08-04 18:00:00')} icon={'01d'} temperature={294.05} selected={true} />);
  expect(container.firstChild.classList.contains('forecast--active')).toBe(true);
  expect(container).toMatchSnapshot();
});
