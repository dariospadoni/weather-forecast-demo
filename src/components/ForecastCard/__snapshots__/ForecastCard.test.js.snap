// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when it is not selected 1`] = `
<div>
  <article
    class="forecast__card "
  >
    <div
      class="forecast__card__time"
    >
      18:00
    </div>
    <div
      class="forecast__card__condition"
    >
      <svg>
        weather-sun.svg
      </svg>
    </div>
    <div
      class="forecast__card__temperature"
    >
      21°
    </div>
  </article>
</div>
`;

exports[`renders correctly when selected 1`] = `
<div>
  <article
    class="forecast__card forecast--active"
  >
    <div
      class="forecast__card__time"
    >
      18:00
    </div>
    <div
      class="forecast__card__condition"
    >
      <svg>
        weather-sun.svg
      </svg>
    </div>
    <div
      class="forecast__card__temperature"
    >
      21°
    </div>
  </article>
</div>
`;
